src/util/ring-fifo.c

```diff
@@ -7,6 +7,18 @@
 
 #include <mgba-util/memory.h>
 
+#if defined _MSC_VER
```

Comment:
Why is this needed? It should already include the file these are defined in.

Reply:
Again, due to rigid nature of win32 interlocked functions. Pointers need different versions of atomic function, but in this module, the code assumes the atomic* version for integers can work for pointers. So I have to do this. It's not best way, but it the quick way without modifying the resting the code to use something like `ATOMIC_STORE_PTR` for example.